challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |string| string.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |string| string == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.select { |string| string != "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input[0..1]
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select { |string| string[0] == "c" || string[0] == "d" }
```

Comment:
Strings have a method [`#start_with?`](http://ruby-doc.org/core-2.2.2/String.html#method-i-start_with-3F) that could make this a bit more readable.